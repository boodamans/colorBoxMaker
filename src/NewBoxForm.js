import React, { useState } from 'react';

function NewBoxForm({ addBox }) {
  const [boxData, setBoxData] = useState({ backgroundColor: '', width: '', height: '' });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setBoxData({ ...boxData, [name]: value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    addBox(boxData);
    setBoxData({ backgroundColor: '', width: '', height: ''});
    console.log(boxData)
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="width">Width:</label>
      <input
        type="number"
        id="width"
        name="width"
        value={boxData.width}
        onChange={handleChange}
      />
      <label htmlFor="height">Height:</label>
      <input
        type="number"
        id="height"
        name="height"
        value={boxData.height}
        onChange={handleChange}
      />
      <label htmlFor="backgroundColor">Background Color:</label>
      <input
        type="text"
        id="backgroundColor"
        name="backgroundColor"
        value={boxData.backgroundColor}
        onChange={handleChange}
      />
      <button type="submit">Create Box</button>
    </form>
  );
}

export default NewBoxForm;
