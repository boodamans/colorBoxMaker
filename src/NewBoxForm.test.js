import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('NewBoxForm component renders without crashing', () => {
  render(<NewBoxForm />);
});

test('NewBoxForm component matches snapshot', () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});

test('NewBoxForm component updates input values and submits the form', () => {
  const { getByLabelText, getByText, getByDisplayValue } = render(
    <NewBoxForm addBox={() => {}} />
  );

  const widthInput = getByLabelText('Width:');
  const heightInput = getByLabelText('Height:');
  const colorInput = getByLabelText('Background Color:');
  const createButton = getByText('Create Box');

  // Test input updates
  fireEvent.change(widthInput, { target: { value: '100' } });
  fireEvent.change(heightInput, { target: { value: '150' } });
  fireEvent.change(colorInput, { target: { value: 'red' } });

  expect(getByDisplayValue('100')).toBeInTheDocument();
  expect(getByDisplayValue('150')).toBeInTheDocument();
  expect(getByDisplayValue('red')).toBeInTheDocument();

  // Test form submission
  fireEvent.click(createButton);
  // Add further tests here based on your application logic
});
