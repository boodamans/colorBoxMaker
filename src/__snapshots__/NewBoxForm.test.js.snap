// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBoxForm component matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      type="number"
      value=""
    />
    <label
      for="backgroundColor"
    >
      Background Color:
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Create Box
    </button>
  </form>
</DocumentFragment>
`;
